libraries/AP_Avoidance/AP_Avoidance.cpp

```diff
@@ -507,10 +514,28 @@ void AP_Avoidance::handle_avoidance_local(AP_Avoidance::Obstacle *threat)
     MAV_COLLISION_THREAT_LEVEL new_threat_level = MAV_COLLISION_THREAT_LEVEL_NONE;
     MAV_COLLISION_ACTION action = MAV_COLLISION_ACTION_NONE;
 
+    // Get vehicle location for altitude
+    Location my_loc;
+    _ahrs.get_position(my_loc);
+
     if (threat != nullptr) {
         new_threat_level = threat->threat_level;
         if (new_threat_level == MAV_COLLISION_THREAT_LEVEL_HIGH) {
-            action = (MAV_COLLISION_ACTION)_fail_action.get();
+            if ((!_fail_altitude_minimum) || (my_loc.alt > _fail_altitude_minimum)) {
```

Comment:
this line should be to:
if ((!_fail_altitude_minimum) || (_ahrs.get_position(my_loc) && my_loc.alt > _fail_altitude_minimum))

and since that's the only place my_loc is used then it could be defined right there above it